frontend/src/views/Org/MembersPage/components/OrgIdentityTab/components/IdentitySection/IdentityAuthMethodModal.tsx

```diff
@@ -86,187 +98,123 @@ export const IdentityAuthMethodModal = ({ popUp, handlePopUpOpen, handlePopUpTog
   });
 
   useEffect(() => {
-    // reset form on open
-    if (popUp.identityAuthMethod.isOpen)
+    if (popUp.identityAuthMethod.isOpen) {
       reset({ authMethod: popUp?.identityAuthMethod?.data?.authMethod });
+    }
   }, [popUp.identityAuthMethod.isOpen]);
 
+  const watchedAuthMethod = watch("authMethod");
+
   const identityAuthMethodData = {
     identityId: popUp?.identityAuthMethod.data?.identityId,
     name: popUp?.identityAuthMethod?.data?.name,
-    authMethod: watch("authMethod")
+    authMethod: watch("authMethod"),
+    configuredAuthMethods: popUp?.identityAuthMethod?.data?.allAuthMethods
   } as {
     identityId: string;
     name: string;
     authMethod?: IdentityAuthMethod;
+    configuredAuthMethods?: IdentityAuthMethod[];
   };
 
+  const isSelectedAuthAlreadyConfigured =
+    identityAuthMethodData?.configuredAuthMethods?.includes(watchedAuthMethod);
+
   useEffect(() => {
-    if (identityAuthMethodData?.authMethod) {
-      setValue("authMethod", identityAuthMethodData.authMethod);
-      return;
+    if (popUp?.identityAuthMethod?.data?.authMethod) {
```

Comment:
If you keep the modal outside. That is `<Modal>...` wrapping outside this will unmount the component. Thus you won't need this useEffect at all. Actually both useEffect can then be removed